skeleton/src/main/java/org/killbill/commons/skeleton/metrics/TimedResourceInterceptor.java

```diff
@@ -29,33 +43,133 @@
  */
 public class TimedResourceInterceptor implements MethodInterceptor {
 
-    private final TimedResourceMetric timer;
+    private final Provider<GuiceContainer> jerseyContainer;
+    private final Provider<MetricRegistry> metricRegistry;
+    private final String resourcePath;
```

Comment:
The interceptor is created by the listener. The listener itself is create during Guice "configuration time" (??), when the actual instances of those classes are not yet available.
See this line: https://github.com/killbill/killbill-commons/pull/16/files#diff-9da97a8842ac4154b2c4c4ee8a5cb529R84

I've based this impl in the existing one, which was using Guice `SettableFuture`. I changed it to using `Provider`, since its api is a little bit easier to use.

I've tried using direct injection, but it would just give me `null`. Maybe there's some way around it?